### PR TITLE
Feature resize4D

### PR DESCRIPTION
**Pull request by Hanyonggong**
This is realize Resize4d inherit Resize2d。

command：
$ cd buddy-mlir/build
$ cmake -G Ninja .. -DBUDDY_EXAMPLES=ON -DBUDDY_ENABLE_OPENCV=ON
$ ninja resize4D
$ cd bin
$ ./resize4D ../../examples/images/YuTu.png result-dip-resize.png

### CODE REVIEW

**Comment by Hanyonggong** [comment]
@Hanyonggong could you please review it

File: examples/DIPDialect/resize4D.cpp [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+#include <buddy/Core/Container.h>
+#include <buddy/DIP/DIP.h>
+#include <buddy/DIP/ImageContainer.h>
+#include <iostream>
+#include <math.h>
+
+using namespace std;
+
+void testImplementation(int argc, char *argv[]) {
+  // Read as grayscale image.
```
Comment (zhanghb97) [inline examples/DIPDialect/resize4D.cpp:34]: Wrong comment: "grayscale image"

File: examples/DIPDialect/resize4D.cpp
Comment (zhanghb97) [inline examples/DIPDialect/resize4D.cpp:1]: Please use `clang-format` to format the code.

File: midend/lib/Conversion/LowerDIP/LowerDIPPass.cpp
Comment (zhanghb97) [inline midend/lib/Conversion/LowerDIP/LowerDIPPass.cpp:1]: Please use `clang-format` to format the code.

File: midend/lib/Utils/DIPUtils.cpp
Comment (zhanghb97) [inline midend/lib/Utils/DIPUtils.cpp:1]: Please use `clang-format` to format the code.

**Review by zhanghb97 — COMMENTED** [review]
Giving quick feedback. Please use `clang-format` to format all your `.cpp/.h` files.
Additionally, please add unit test for your operation and vectorization pass.

File: tests/Dialect/DIP/resize4D_lowering.mlir
Comment (zhanghb97) [inline tests/Dialect/DIP/resize4D_lowering.mlir:1]: This test case does not test the `lower-dip` pass.

**Review by zhanghb97 — APPROVED** [review]
I merged this PR first; please modify the lowering test in the next PR.

**Pull request merged**